### PR TITLE
Modify to use terraform 0.12

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
Initial work to change this to work with 0.12

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -463,12 +467,12 @@ resource "aws_ecs_service" "main_no_lb" {
   deployment_minimum_healthy_percent = "${var.tasks_minimum_healthy_percent}"
   deployment_maximum_percent         = "${var.tasks_maximum_percent}"
 
-  ordered_placement_strategy = "${local.ecs_service_ordered_placement_strategy[local.ecs_service_launch_type]}"
-  placement_constraints      = "${local.ecs_service_placement_constraints[local.ecs_service_launch_type]}"
+  ordered_placement_strategy = local.ecs_service_ordered_placement_strategy[local.ecs_service_launch_type]
+  placement_constraints = local.ecs_service_placement_constraints[local.ecs_service_launch_type]
```
Comment (chrisgilmerproj) [inline main.tf]: This doesn't work yet.

File: main.tf [hunk 2]
```diff
@@ -423,12 +423,16 @@ resource "aws_ecs_service" "main" {
...
   deployment_maximum_percent         = "${var.tasks_maximum_percent}"
 
-  ordered_placement_strategy = "${local.ecs_service_ordered_placement_strategy[local.ecs_service_launch_type]}"
-  placement_constraints      = "${local.ecs_service_placement_constraints[local.ecs_service_launch_type]}"
+  ordered_placement_strategy {
+    local.ecs_service_ordered_placement_strategy[local.ecs_service_launch_type]
+  }
+  placement_constraints {
+    local.ecs_service_placement_constraints[local.ecs_service_launch_type]
+  }
```
Comment (chrisgilmerproj) [inline main.tf]: This doesn't work yet.

File: main.tf (hunk 1 above)
Comment (cblkwell) [reply main.tf]: It works now!

File: main.tf (hunk 2 above)
Comment (cblkwell) [reply main.tf]: Also works now.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -17,8 +17,7 @@ repos:
       - id: markdownlint
 
   - repo: git://github.com/antonbabenko/pre-commit-terraform
-    rev: v1.12.0
+    rev: v1.17.0
     hooks:
       - id: terraform_docs
       - id: terraform_fmt
-      - id: terraform_validate_no_variables
```
Comment (dynamike) [inline .pre-commit-config.yaml]: why did we remove terraform validate?

File: README.md [hunk 4]
```diff
@@ -1,65 +1,4 @@
 <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
-Creates an ECS service.
```
Comment (dynamike) [inline README.md]: This also looks like terraform-docs in generating useful documentation.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (cblkwell) [reply .pre-commit-config.yaml]: I think it doesn't work with TF12 yet, if I recall...I can try poking it.

File: README.md (hunk 4 above)
Comment (cblkwell) [reply README.md]: Erm...not sure what you're saying. I believe terraform-docs is also broken for TF12. I'll see what I can do...

File: .pre-commit-config.yaml (hunk 3 above)
Comment (chrisgilmerproj) [reply .pre-commit-config.yaml]: I think it was removed in v1.15 and replaced with something else. Also the `terraform_docs` doesn't work right if you happen to not be using gnu-awk.

File: README.md (hunk 4 above)
Comment (chrisgilmerproj) [reply README.md]: terraform-docs changed and also broke for tf12.

**Pull request merged**